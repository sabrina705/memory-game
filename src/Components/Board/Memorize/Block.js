import React from 'react';
import classnames from 'classnames';

const Block = ({ gem, width }) =>
  <div className={classnames('block', { gem })} style={{ width }} />;

export default Block;
