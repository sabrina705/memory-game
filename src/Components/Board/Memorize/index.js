import React, { Component, PropTypes } from 'react';
import { observer, inject } from 'mobx-react';
import queue from 'async/queue';
import Block from './Block';

export class Memorize extends Component {
  static propTypes = {
    gems: PropTypes.object.isRequired,
    blocks: PropTypes.object.isRequired,
    length: PropTypes.number.isRequired
  };

  state = {
    gems: []
  };

  componentDidMount() {
    const { gems } = this.props;

    /** add a gem after a center time */
    const q = queue((gem, done) => {
      this.setState({
        gems: [...this.state.gems, gem]
      }, () => setTimeout(done, 1000));
    }, 1);

    q.drain = (() => {
      this.props.router.push('/play');
    });

    gems.forEach(gem => q.push(gem));
  }

  renderBlock = (block, index) => {
    const { gems } = this.state;

    return <Block
      key={block.id}
      gem={gems.includes(index)}
      width={50}
    />;
  };

  render() {
    const { blocks, length } = this.props;

    return (
      <div className="Memorize" style={{ width: length * 50,  margin: '0 auto' }}>
        { blocks.map(this.renderBlock) }
        <div><center>Memorize</center></div>
      </div>
    );
  }
}


export default inject(store => {
  store.board.generate();
  store.board.draw();

  return ({
    length: store.board.currentLevel.length,
    blocks: store.board.blocks,
    gems: store.board.gems
  });
})(observer(Memorize));
