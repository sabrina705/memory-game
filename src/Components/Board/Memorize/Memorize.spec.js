import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { Memorize } from '../Memorize';
import mobx from 'mobx';

const gems = mobx.observable([1, 2, 3]);
const blocks = mobx.observable([
  { id: 0, gem: false, selected: false },
  { id: 1, gem: true, selected: false },
  { id: 2, gem: true, selected: false },
  { id: 3, gem: true, selected: false },
  { id: 4, gem: false, selected: false }
]);

const router = {
  push() {

  }
};

it('Memorize renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render((
    <Memorize
      router={router}
      gems={gems}
      blocks={blocks}
      length={3}
    />
  ), div);
});

it('Memorize/Block matches snapshot gem true', () => {
  const component = renderer.create(
    <Memorize
      router={router}
      gems={gems}
      blocks={blocks}
      length={3}
    />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
