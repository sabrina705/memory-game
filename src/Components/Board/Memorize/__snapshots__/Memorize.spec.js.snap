exports[`test Memorize/Block matches snapshot gem true 1`] = `
<div
  className="Memorize"
  style={
    Object {
      "width": 150,
    }
  }>
  <div
    className="block"
    style={
      Object {
        "width": 50,
      }
    } />
  <div
    className="block"
    style={
      Object {
        "width": 50,
      }
    } />
  <div
    className="block"
    style={
      Object {
        "width": 50,
      }
    } />
  <div
    className="block"
    style={
      Object {
        "width": 50,
      }
    } />
  <div
    className="block"
    style={
      Object {
        "width": 50,
      }
    } />
</div>
`;
